tern/analyze/docker/container.py

```diff
@@ -192,3 +217,104 @@ def close_client():
         # it should either already be closed, no socker is in use,
         # or docker is not setup -- either way, the socket is closed
         pass
+
+
+def generate_manifest_url(image_name, image_tag):
+    '''
+    Generate a docker image manifest url end point with
+    given image name and tag
+    '''
+    url = "{0}/{1}/manifests/{2}"
+    return url.format(docker_endpoint, image_name, image_tag)
+
+
+def generage_image_blobs_url(image_name, digest):
+    '''
+    Generate a docker image blobs url end point with
+    given image name and tag
+    '''
+    url = "{0}/{1}/blobs/{2}"
+    return url.format(docker_endpoint, image_name, digest)
+
+
+def is_token_expired(auth_token):
```

Comment:
I had implemented this by checking if the response is 401. If it is, then get a fresh token and try again. What do you think of this method?

Reply:
I think the current implementation is much cleaner compared to 401 response.